chore(docs): gatsby-source-wordpress Tutorial Additions (#15967)
gatsbyjs/gatsby: docs/tutorial/wordpress-source-plugin-tutorial.md

    }
  }
`
//highlight-end
```

Save these changes and look at localhost:8000 to see your new homepage with list of sorted blog posts!

![WordPress home after query](/images/wordpress-source-plugin-home.jpg)

> **NOTE:** to future editors: it would be useful to also have examples of how to load blog posts to their own individual pages. And helpful to insert a screenshot of the final result here

## Create pages for each blog post and link to them

An index page with a post title and excerpt is great, but we should also build pages out for each of the blog posts, and link to them from our `index.js` file.

To do this, we need to:

1. Create pages for each blog post
2. Link up the title on the index page with the post page.

> If you haven't already, please read through [Part 7](/tutorial/part-seven/) of the foundational tutorial, as it goes through the concept and examples of this process.

Reviewer: ```suggestion
If you haven't already, please read through [Part 7](/tutorial/part-seven/) of the foundational tutorial, as it goes through the concept and examples of this process with Markdown instead of WordPress.
```

I think pointing out that this is supposed to be specifically for WordPress would help understand why you'd follow this tutorial instead of just relying on the one in the Tutorial section.

Author: Good point!

### Creating pages for each blog post.

> In Part 7 of the tutorial, the first step in creating pages in creating slugs for the markdown files. Since we are using WordPress and not Markdown files, we can grab the slugs that get returned from our API call to the WordPress source. We can skip creating slugs, since we already have them.

Reviewer: ```suggestion
In Part 7 of the tutorial, the first step in creating pages is creating slugs for the markdown files. Since we are using WordPress and not Markdown files, you can grab the slugs that get returned from your API call to the WordPress source. You can skip creating slugs, since you already have them.
```

![Two posts logged to the terminal](/images/wordpress-source-plugin-log.jpg)

Excellent! As explained in Part 7 of the tutorial, this `createPages` export is one of the Gatsby "workhorses" and allows us to create our blog posts (or pages, or custom post types, etc.) from our WordPress install.

Before we can create the blog posts, however, we need to specify a template to build the pages.

In your `src` directory, create a directory called `templates` and in the newly created `templates` folder, create a filed named `blog-posts.js`. In that new file, paste the following:

```jsx:title=src/tempates/blog-posts.js
import React from "react"
import Layout from "../components/layout"
import { graphql } from "gatsby"

export default ({ data }) => {
  const post = data.allWordpressPost.edges[0].node
  console.log(post)
  return (
    <Layout>
      <div>
        <h1>{post.title}</h1>
        <div dangerouslySetInnerHTML={{ __html: post.content }} />
      </div>
    </Layout>
  )
}
export const query = graphql`
  query($slug: String!) {
    allWordpressPost(filter: { slug: { eq: $slug } }) {
      edges {
        node {
          title
          content
        }
      }
    }
  }
`
```

> What is this file doing? After importing our dependencies, we construct the layout of the post with JSX. We wrap everything in the `Layout` component, so our style is the same throughout the site. Then, we simply add the post title and the pot content. You can add anything you want and can query for here (e.g. feature image, post meta, custom fields, etc.).

Reviewer: ```suggestion
What is this file doing? After importing our dependencies, we construct the layout of the post with JSX. We wrap everything in the `Layout` component, so our style is the same throughout the site. Then, we simply add the post title and the post content. You can add anything you want and can query for here (e.g. feature image, post meta, custom fields, etc.).
```
